File: Navidad 2024/2. Framing names/main.js
function createFrame(names) {
    let max = 0;
    let lines = [];
    let frame = "";

    for (let i = 0; i < names.length; i++) {
        if (names[i].length > max) {
            max = names[i].length;
        }
        lines.push(names[i]);
    }

    for (let i = 0; i < lines.length; i++) {
        if (lines[i].length < max) {
            lines[i] = lines[i] + " ".repeat(max - lines[i].length);
        }
        lines[i] = "* " + lines[i] + " *\n";
    }

    max += 4;

    frame = lines.join("");
    frame = "*".repeat(max) + "\n" + frame + "*".repeat(max);

    return frame;
}

console.log(createFrame(["midu", "madeval", "educalvolpz"]));
console.log(createFrame(["midu"]));
console.log(createFrame(["a", "bb", "ccc"]));

File: Navidad 2024/2. Framing names/main.py
def createFrame(names):
    # Declaramos las variables a usar
    max = 0
    lines = []
    frame = ""
    i = 0
    
    # Bucle para determinar el string con la mayor longitud
    for name in names:
        if len(name) > max:
            max = len(name)
        
        lines.append(name)
    
    # Bucle para modificar cada linea de los nombres y adaptarlo al cuadro
    while i < len(lines):
        if len(lines[i]) < max:
            lines[i] = lines[i] + (" " * (max - len(lines[i])))
        
        lines[i] = "* " + lines[i] + " *\n"
        
        i = i + 1
    
    # Ajustes para el frame
    max += 4
    frame = "".join(lines)
    frame = "*" * max + "\n" + frame + "*" * max
    
    return frame

print(createFrame(['midu', 'madeval', 'educalvolpz']))